### PR TITLE
Kafka encryption

### LINKED ISSUE #314: Kafka Encryption
https://miro.com/app/board/o9J_kn9c9wc=/

### PR DESCRIPTION
**Pull request by ldklenner**
### Reason for this PR
- #135 
- #314 

### Changes in this PR
- Added Kafka encryption layer through KafkaEncryptionUtility and KafkaEncryptionComponent
- Get secrets from config
- Added SecretsUtility for key derivation
- Closes #314 

- [x] Test written
- [x] Changelog updated

### CODE REVIEW

**Comment by ldklenner** [comment]
Do not merge until further security review

File: product_code/authentication_service/impl/src/main/resources/prod-application.conf
Comment (david-buderus) [inline product_code/authentication_service/impl/src/main/resources/prod-application.conf:14]: Why is there a salt group?

File: product_code/certificate_service/impl/src/main/resources/application.conf
Comment (david-buderus) [inline product_code/certificate_service/impl/src/main/resources/application.conf:20]: Why is there a salt group?

File: product_code/certificate_service/impl/src/main/resources/prod-application.conf
Comment (david-buderus) [inline product_code/certificate_service/impl/src/main/resources/prod-application.conf:18]: Why is there a salt group?

File: product_code/user_service/impl/src/main/resources/application.conf
Comment (david-buderus) [inline product_code/user_service/impl/src/main/resources/application.conf:26]: Why is there a salt group?

File: product_code/user_service/impl/src/main/resources/prod-application.conf
Comment (david-buderus) [inline product_code/user_service/impl/src/main/resources/prod-application.conf:14]: Why is there a salt group?

**Review by david-buderus — CHANGES_REQUESTED** [review]

File: product_code/authentication_service/impl/src/main/resources/prod-application.conf
Comment (lrschaef) [reply product_code/authentication_service/impl/src/main/resources/prod-application.conf:14]: To make the config file a bit more readable. The group is used to provide a separator between the secrets and the salts. Of course it looks weird, because we only have one salt atm and we may change this if we only need one salt.

**Review by RHelmert — APPROVED** [review]

**Review by david-buderus — APPROVED** [review]

**Review by dhelmert — APPROVED** [review]

**Pull request merged**
